Project-Hotpot/Project-Hotpot/Parse-Backend/PFPlaylist.swift

```diff
@@ -100,33 +85,40 @@ class PFPlaylist: PFObject, PFSubclassing {
         }
     }
     
-    class func getAllSongsFromPlaylist(playlist: PFPlaylist, completion: @escaping ([PFSong]?, Error?)->Void){
+    class func getAllSongsFromPlaylist(playlist: PFPlaylist, completion: @escaping ([PFSong]?, Error?)->Void) {
         let query = PFQuery(className: "SongJoinTable")
-        var songArray: [PFSong]? = []
-        //playlist is a pointer to playlist object
         query.includeKey("song")
         query.whereKey("playlist", equalTo: playlist)
         query.order(byAscending: "addedAt")
         query.findObjectsInBackground { objects, error in
+            var songArray: [PFSong] = []
+            if let error = error{
+                completion(nil, error)
+                NSLog("error occured: \(error)")
+                return
+            }
+            else if (objects?.isEmpty == true) {
+                NSLog("No results found")
+            }
             //TODO: is there a more efficient way to do this? similar to O(n) efficiency.
-            if let objects = objects{
-                for o in objects{
+            else if let objects = objects {
+                //TODO: switch to inbuilt function such as filter or map
+                for o in objects {
                     if let songObject = o.object(forKey: "song") as? PFSong{
-                        songArray?.append(songObject)
+                        songArray.append(songObject)
                     }
                 }
                 completion(songArray, nil)
             }
-            else{
-                completion(nil, error)
-                NSLog("didn't find anything")
-            }
         }
     }
     
-    class func getAllPlaylistsInBackground(completion: @escaping([PFPlaylist]?, Error?) -> Void) {
+    class func getNPlaylistsInBackground(limit: Int?, completion: @escaping([PFPlaylist]?, Error?) -> Void) {
```

Comment:
May want to think about the name a bit. To the caller looks like it will fetch n playlists. Could refine more to depict that you are getting 'N' latest created playlists